### PR TITLE
fix the logic of manager expectations feature 

### PR DESCRIPTION
**Pull request by TarekIdrees**
fix the logic of the code and apply some changes to test cases by increase the amount of data

### CODE REVIEW

**Review by mosayed01 — DISMISSED** [review]

File: src/main/kotlin/interactor/GetCityThatMatchTheManagerExpectationsInteractor.kt [hunk 1]
```diff
@@ -1,61 +1,40 @@
...
+            .takeIf { it.isNotEmpty() }
+            ?.run {
+                excludeNullPricesOfMeals()
+               .getCityThatMatchExpectations()
+            } ?: throw IllegalStateException("List of cities is empty")
     }
 
-    private fun excludeNullPricesOfMeals(city: CityEntity): Boolean {
-        return city.mealsPrices.mealFor2PeopleMidRangeRestaurant != null
+    private fun List<CityEntity>.excludeNullPricesOfMeals() = filter {
```
Comment (AmeerAmjed) [inline src/main/kotlin/interactor/GetCityThatMatchTheManagerExpectationsInteractor.kt]: here not need extension function can put filter in run scope and move block this function into model "MealsPrices" .

Comment (AmeerAmjed) [reply src/main/kotlin/interactor/GetCityThatMatchTheManagerExpectationsInteractor.kt]: not use **compact function** same this case.

File: src/test/kotlin/interactor/GetCityThatMatchTheManagerExpectationsInteractorTest.kt [hunk 2]
```diff
@@ -1,87 +1,124 @@
...
+            interactor.execute(countries).cityName
+        }
         // Then check the final result
-        assertNull(null, result?.cityName)
+        assertThrows(IllegalStateException::class.java, actualData)
     }
 
+    companion object{
+        // Given the countries which will search on it
+        val countries = listOf("united states", "canada", "mexico")
```
Comment (AmeerAmjed) [inline src/test/kotlin/interactor/GetCityThatMatchTheManagerExpectationsInteractorTest.kt:131]: when use variable just in same class set private variable

**Review by AmeerAmjed — APPROVED** [review]

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]
apply changed

File: src/test/kotlin/interactor/GetCityThatMatchTheManagerExpectationsInteractorTest.kt (hunk 2 above)
Comment (TarekIdrees) [reply src/test/kotlin/interactor/GetCityThatMatchTheManagerExpectationsInteractorTest.kt:131]: yes sure

**Review by AmeerAmjed — DISMISSED** [review]

**Review by mosayed01 — DISMISSED** [review]

File: src/main/kotlin/interactor/GetCityThatMatchTheManagerExpectationsInteractor.kt [hunk 3]
```diff
@@ -1,61 +1,38 @@
...
 
 class GetCityThatMatchTheManagerExpectationsInteractor(
-    private val dataSource: CostOfLivingDataSource
+    private val dataSource: CostOfLivingDataSource,
 ) {
 
-    fun execute(countries: List<String>): CityEntity? {
-        val existCountries = countries.map(::formatCountriesNames)
-            .filter { selectedCountries.contains(it) }
+    fun execute(selectedCountries: List<String>): CityEntity {
```
Comment (msalarab) [inline src/main/kotlin/interactor/GetCityThatMatchTheManagerExpectationsInteractor.kt]: change to invoke

**Review by msalarab — CHANGES_REQUESTED** [review]

File: src/main/kotlin/utils/Helper.kt [hunk 4]
```diff
@@ -88,7 +88,7 @@ class Helper(private val dataSource: CostOfLivingDataSource) {
                     print("example iraq, usa:-")
                     val countries = readlnOrNull().toString().split(",")
                     if (countries.isNotEmpty()) {
-                        print(GetCityThatMatchTheManagerExpectationsInteractor(dataSource).execute(countries))
+                        print(GetCityThatMatchTheManagerExpectationsInteractor(dataSource).invoke(countries))
```
Comment (msalarab) [inline src/main/kotlin/utils/Helper.kt]: remove invoke

Comment (msalarab) [reply src/main/kotlin/utils/Helper.kt]: @AmeerAmjed

**Review by msalarab — DISMISSED** [review]

**Review by mosayed01 — DISMISSED** [review]

**Review by msalarab — DISMISSED** [review]

**Review by AmeerAmjed — APPROVED** [review]

**Pull request merged**